### PR TITLE
support for createCipher backward compatible

### PR DESCRIPTION
**Pull request by juanpicado**
ref https://github.com/verdaccio/verdaccio/issues/4601

### CODE REVIEW

File: packages/signature/src/index.ts [hunk 1]
```diff
@@ -2,7 +2,10 @@ export {
   aesDecryptDeprecated,
   aesEncryptDeprecated,
```
Comment (juanpicado) [inline packages/signature/src/index.ts:3]: I could entirely remove this function if make sense.

**Comment by mbtools** [comment]
The code looks good 👍  I don't have a legacy setup to test if this decodes existing tokens correctly 🤷‍♂️ 

Sometimes, I think "we" (open source communities) are too kind and keep supporting things that really should not be used anymore. Node took the legacy feature away for a reason (after years of notice). Anyone upgrading node should be strongly encouraged to set up their security properly as well.

**Comment by juanpicado** [comment]
> The code looks good 👍  I don't have a legacy setup to test if this decodes existing tokens correctly 🤷‍♂️ 
> 
> 
> 
> Sometimes, I think "we" (open source communities) are too kind and keep supporting things that really should not be used anymore. Node took the legacy feature away for a reason (after years of notice). Anyone upgrading node should be strongly encouraged to set up their security properly as well. 

I agree, but try to follow a degradation approach of features to minimize breaking changes, I will follow up making some changes in 5.x to encouraging current users (old installations) move to modern signature and new installations will use it by default, also disable the old support in 6.x by default with an option to backward to in cases someone need it with a big warning that should not be used, after 6.x will goes away indefinitely. In theory most of the tokens are used locally so should be fine, furthermore user always have had the chance to use JWT tokens if they consider legacy is not secure enough.

**Comment by juanpicado** [comment]
My test were incomplete, turns out does not work, the function is not able to decode an already existing token, thus is not backward compatible.

**Comment by mbtools** [comment]
Did you try with KEY_SIZE = 32 ?

**Comment by juanpicado** [comment]
> KEY_SIZE

Yep, I always get `RangeError: Invalid key length`. Anyhow thanks for the idea, alternatively  I've been working in a different strategy, force progressively upgrades if users goes to Node.js 22. https://github.com/verdaccio/verdaccio/pull/4621/files

**Pull request merged**